### PR TITLE
Restrict patching of read-only bios attributes

### PR DESCRIPTION
**Pull request by asmithakarun**
Since the P10 system has BMC to manage the BIOS attributes,
and bmcweb does not know the attribute values of the BIOS,
it should allow the host to be able to set BIOS readOnly
attributes, while the redfish-doPATCH operation should not
be allowed.

With this change, when a redfish command to update a bios
attribute (which is read-only), is fired, bmcweb will send
the error back to the client.

Tested By:

Updation of read-only attr:

-X PATCH https://${bmc}/redfish/v1/Systems/system/Bios/Settings -d '{"Attributes":{"hb_debug_console":"Disabled"}}'
{
  "hb_debug_console@Message.ExtendedInfo": [
    {
      "@odata.type": "#Message.v1_1_1.Message",
      "Message": "The property hb_debug_console is a read only property and cannot be assigned a value.",
      "MessageArgs": [
        "hb_debug_console"
      ],
      "MessageId": "Base.1.8.1.PropertyNotWritable",
      "MessageSeverity": "Warning",
      "Resolution": "Remove the property from the request body and resubmit the request if the operation failed."
    }
  ]
}

Signed-off-by: Asmitha Karunanithi <asmitk01@in.ibm.com>

### CODE REVIEW

**Comment by geissonator** [comment]
add to approvelist

File: redfish-core/lib/bios.hpp [hunk 1]
```diff
@@ -445,7 +448,16 @@ inline void requestRoutesBiosSettings(App& app)
                             return;
                         }
 
-                        std::string biosAttrType = (*it).second;
+                        bool biosAttrReadOnlyStatus = ((*it).second).first;
+                        if (biosAttrReadOnlyStatus == true)
+                        {
+                            BMCWEB_LOG_ERROR
+                                << "Attribute Type is ReadOnly. Patch failed!";
+                            messages::internalError(asyncResp->res);
```
Comment (gtmills) [inline redfish-core/lib/bios.hpp:456]: I don't think this should be internal Error, how about PropertyNotWritable ?
https://redfish.dmtf.org/registries/Base.1.10.0.json

**Review by sunharis — CHANGES_REQUESTED** [review]
Please return the error : PropertyNotWritable

File: redfish-core/lib/bios.hpp (hunk 1 above)
Comment (asmithakarun) [reply redfish-core/lib/bios.hpp]: done

**Comment by gtmills** [comment]
Upstream for this?

**Comment by asmithakarun** [comment]
> Please return the error : PropertyNotWritable

Done

**Comment by geissonator** [comment]
CI fail is due to clang format:
```
-                            messages::propertyNotWritable(asyncResp->res, attrName);
+                            messages::propertyNotWritable(asyncResp->res,
+                                                          attrName);
```

**Review by gtmills — CHANGES_REQUESTED** [review]
CI fail

**Comment by gtmills** [comment]
Upstream for this?

**Review by gtmills — APPROVED** [review]
Seems good to me. Can we get this upstream?

**Comment by asmithakarun** [comment]
This is the upstream change:
https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/38702

I will push this change to the upstream Gerrit commit.
Seems the parent commits doesn't contain the changes for node class migration..

**Review by sunharis — APPROVED** [review]

**Pull request merged**